gis/text_to_centreline_geometry_scripts/README.md

```diff
@@ -0,0 +1,189 @@
+4. dissolve the centreline segments for each community safety zone into one geom
+5.
+ - **High level:** cut the dissolved segment to either be 
+    1. Intersection 1 (`btwn1` intersection) to  translated intersection 2 (`btwn2` intersection)
+    
+    ![](jpg/case1_num1.jpg)
+ 
+    2. Translated intersection 1 (`btwn1` intersection) to intersection 2 (`btwn2` intersection)
+    
+    ![](jpg/case1_num2.jpg)
+
+    3. Translated intersection 1 (`btwn1` intersection)  to translated intersection 2 (`btwn2` intersection)
+    
+    ![](jpg/case1_num3.jpg)
+
+- The [ST_LineLocatePoint function](https://postgis.net/docs/ST_LineLocatePoint.html) was used to find the location of points as a fraction of the length of the dissolved centreline geom.
+ - I used the [ST_LineSubstring function](ST_LineSubstring), which takes a geom and two numbers between 0 and 1 inclusive, and returns a substring of that line geom from the first fraction to the second fraction (i.e. if the fractions were 0 and 0.20 `ST_LineSubstring` would return the first 20% of the inputted line geom). To find the fraction that the point x metres away from the original intersection was located, I calculated the fraction of the line that the location the specified intersection represented. I then added or subtracted (depending on direction and location of the intersection on the dissolved centreline geom). One of my concerns was: **how do I know if I should add or subtract the percentage of x metres of the line?**
+ - In order to figure out if I should add or subtract the fraction of the line that is x metres away from the intersection, I used the line from step 1 (step 1 of special case 2 workflow). I found the closest point on the dissolved centreline to the end of the line created in step 1.
+ - If the fraction returned by ST_LineLocatePoint of the dissolved centreline that the original intersection occurred on was smaller than the fraction that returned by ST_LineLocatePoint represented the closest point on the dissolved centreline to the end of the line from step 1, then the situation would look like this (assuming that the line goes from left to right so points on the left have a lower fraction value):
+![](jpg/figure1_case2.jpg)
+ - In the situation in the above image, to get the fraction of the point x metres away from the original intersection, you would add the `x metres/total length of dissolved centreline` to the fraction returned by `ST_LineLocatePoint` of the original intersection.
+ ![](jpg/figure2_case2.jpg)
+ - In the situation in the above image (assuming the line is drwan from left to right so that the fractions from St_LineLocatePoint of the points on the left are smaller), to get the fraction of the point x metres away from the original intersection, you would subtract the `x metres/total length of dissolved centreline` to the fraction returned by LineLocatePoint of the original intersection.
+
+# MultiLineString Geometries that are Case 2
```

Comment:
Are offset streets only an issue with bylaws where at least one point is offset from an intersection?

Reply:
Also this should be an `##` level header or convert it to an issue?